tools/Azure.Mcp.Tools.Foundry/src/Commands/FoundryJsonContext.cs

```diff
@@ -56,5 +56,18 @@ namespace Azure.Mcp.Tools.Foundry.Commands;
 [JsonSerializable(typeof(AgentsQueryAndEvaluateResult))]
 [JsonSerializable(typeof(AgentsEvaluateResult))]
 [JsonSerializable(typeof(CognitiveServicesAccountDeploymentData))]
+[JsonSerializable(typeof(Dictionary<string, object>))]
+[JsonSerializable(typeof(Dictionary<string, string>))]
```

Comment:
my understanding is that using `[JsonSerializable(typeof(...<object>))]` (i.e. object)  is not ideal for trimming, Each value is object, so at runtime the serializer needs to discover the actual type of every value. That adds overhead and impact many source-gen optimizations.

I remember replacing object with the more trim-safe `JsonElement` in some services. Does that approach works  here? 

Tagging @agocke if he has some thoughts.

Reply:
I believe the reason for the serialization error is that the code is trying to obtain type information for our local type `ToolDefinitionAIFunction` from the JSON context of an external type (`Microsoft.Extensions.AI.AIJsonUtilities.DefaultOptions.GetTypeInfo`), which has no knowledge of it. Ideally, the fix would be to define our own JSON context for `ToolDefinitionAIFunction`, like we do for other models.